src/Components/Reviews.jsx

```diff
@@ -1,27 +1,39 @@
-import { getReviews } from "../api"
+import { getReviews, getReviewsByCategory } from "../api"
 import { useEffect, useState } from 'react'
-import { Link } from "react-router-dom"
+import { Link, useSearchParams } from "react-router-dom"
+
 
 const Reviews = () => {
     const [reviewList, setReviews] = useState([])
     const [isLoading, setIsLoading] = useState(true)
+    const [searchParams, setSearchParams] = useSearchParams()
+    const sortByQuery= searchParams.get('category')
+    
 
     useEffect(()=>{
         setIsLoading(true)
-        getReviews()
-        .then((reviews)=>{
-            setReviews(reviews)
-            setIsLoading(false)
-        })
-    }, [])
+        if(sortByQuery===null){
+            getReviews()
+            .then((reviews)=>{
+                setReviews(reviews)
+                setIsLoading(false)
+            })
+        } else{
+            getReviewsByCategory(sortByQuery)
+            .then((reviews)=>{
+                setReviews(reviews)
+                setIsLoading(false)
+            })
```

Comment:
You could make this less DRY by using the getReviews function and passing a query to it. It will either use the query if it's there or not if it isn't. Should be reusable